File: extensions/services/pg2luciddb/samples/perl/fetchdata.pl
#!perl

use DBI;
use DBD::Pg;

### change settings to your host with pg2luciddb:
my $dbname = "LOCALDB";
my $host = "localhost";
my $port = 9999;
my $dbuser = "sa";
my $dbpass = "";

$dbh = DBI->connect("dbi:Pg:dbname=$dbname;host=$host;port=$port", $dbuser, $dbpass, {AutoCommit => 0, RaiseError => 1, PrintError => 0});

print "Established connection to PG2LucidDB\n";

### get list of locations:
print "\nLOCATIONS table\n";
print "ID	NAME	MANAGER	SQUARE	ISFASTFOOD	ISCLOSED	DATEADDED	DATEOPENED\n\n";	

my $SQL = "SELECT * FROM PG2LUCIDDBTEST.LOCATION ORDER BY ID";

for (@{$dbh->selectall_arrayref($SQL)}) 
{
     print $_->[0] . "\t" . $_->[1] . "\t" . $_->[2] . "\t" . $_->[3] . "\t" . $_->[4] . "\t" . $_->[5] . "\t" . $_->[6] . "\t" . $_->[7] . "\n";
}

### get list of SKUs:
print "\n\nSKU table\n";
print "ID	NAME	PRICE	CURRENCY\n\n";

$SQL = "SELECT ID, NAME, PRICE, CURRENCY FROM PG2LUCIDDBTEST.SKU ORDER BY ID";

for (@{$dbh->selectall_arrayref($SQL)}) 
{
     print $_->[0] . "\t" . $_->[1] . "\t" . $_->[2] . "\t" . $_->[3] . "\n";
}

### get sales fact:
print "\n\nSALES_FACT for the date 2009-09-23\n";
print "TRANSACTION_ID	LOCATION	SKU	TIME	DATE\n\n";
$dbh->do("SET SCHEMA 'PG2LUCIDDBTEST'");

$SQL = "SELECT T1.TRANSACTIONID, T2.NAME, T3.NAME, T1.TRANSACTIONTIME, T1.TRANSACTIONDATE FROM SALES_FACT T1 INNER JOIN LOCATION T2 ON T2.ID = T1.LOCATIONID INNER JOIN SKU T3 ON T3.ID = T1.SKUID WHERE T1.TRANSACTIONDATE = APPLIB.CHAR_TO_DATE('yyyy-M-d', '2009-09-23') ORDER BY T1.TRANSACTIONID";

for (@{$dbh->selectall_arrayref($SQL)}) 
{
     print $_->[0] . "\t" . $_->[1] . "\t" . $_->[2] . "\t" . $_->[3] . "\t" . $_->[4] . "\n";
}

$dbh->disconnect();

File: extensions/services/pg2luciddb/samples/php/fetchdata.php
<?php

function print_table($result)
{
    $numrows = pg_num_rows($result);
    $fnum = pg_num_fields($result);

    for ($x = 0; $x < $fnum; $x++) {
        echo strtoupper(pg_field_name($result, $x));
        echo "\t";
    }

    echo "\n\n";

    for ($i = 0; $i < $numrows; $i++) 
    {
        $row = pg_fetch_object($result, $i);

        for ($x = 0; $x < $fnum; $x++) 
        {
            $fieldname = pg_field_name($result, $x);    
    	    echo $row->$fieldname . "\t";
        }
        echo"\n";
    }
}

 // TODO: change your connection string here
 $connectionString = "host=localhost port=9999 dbname=LOCALDB user=sa password=";

 // open connection to the server
 $connection = pg_connect($connectionString) or die ("Unable to connect to PG2LucidDB: " . pg_last_error($connection)); 

 echo("Established connection to PG2LucidDB\n");

 // get list of locations:
 echo("\nLOCATIONS table\n");
 $SQL = "SELECT * FROM PG2LUCIDDBTEST.LOCATION ORDER BY ID";

 $result = pg_query($connection, $SQL);
 print_table($result);
 pg_free_result($result);

 // get list of SKUs:
 echo("\n\nSKU table\n");  
 $SQL = "SELECT ID, NAME, PRICE, CURRENCY FROM PG2LUCIDDBTEST.SKU ORDER BY ID";

 $result = pg_query($connection, $SQL);
 print_table($result);
 pg_free_result($result);
  
 // get sales fact:
 echo("\n\nSALES_FACT for the date 2009-09-23\n");
 pg_query($connection, "SET SCHEMA 'PG2LUCIDDBTEST'");
 $SQL = "SELECT T1.TRANSACTIONID, T2.NAME, T3.NAME, T1.TRANSACTIONTIME, T1.TRANSACTIONDATE FROM SALES_FACT T1 INNER JOIN LOCATION T2 ON T2.ID = T1.LOCATIONID INNER JOIN SKU T3 ON T3.ID = T1.SKUID WHERE T1.TRANSACTIONDATE = APPLIB.CHAR_TO_DATE('yyyy-M-d', '2009-09-23') ORDER BY T1.TRANSACTIONID";
 
 $result = pg_query($connection, $SQL);
 print_table($result);
 pg_free_result($result);

 // close connection:
 pg_close($connection);  

?>